walletnode/api/services/multipart.go

```diff
@@ -77,3 +82,178 @@ func UploadImageDecoderFunc(ctx context.Context, service *Artwork) server.Artwor
+					res.PrimaryLanguage = &value
+				case "native_currency":
+					value := string(buffer)
+					res.NativeCurrency = &value
+				case "realname":
+					value := string(buffer)
+					res.RealName = &value
+				case "twitter_link":
+					value := string(buffer)
+					res.TwitterLink = &value
+				}
+			} else {
+				// Process for the field that have a files
+				filename := part.FileName()
+				filePart := new(bytes.Buffer)
+				if _, err := io.Copy(filePart, part); err != nil {
+					return userdatas.MakeInternalServerError(errors.Errorf("failed to write data to %q: %w", filename, err))
+				}
+
+				switch part.FormName() {
+				case "avatar_image":
+					res.AvatarImage = &userdatas.UserImageUploadPayload{
+						Content:  filePart.Bytes(),
+						Filename: &filename,
+					}
+				case "cover_photo":
+					res.CoverPhoto = &userdatas.UserImageUploadPayload{
+						Content:  filePart.Bytes(),
+						Filename: &filename,
+					}
+				}
+				log.WithContext(ctx).Debugf("Multipart process image: %q", filename)
+			}
+		}
+
+		response = &res
+
+		*p = response
+		return nil
+	}
+}
+
+// UserdatasUpdateUserdataDecoderFunc implements the multipart decoder for service "userdatas" endpoint "/update".
+// The decoder must populate the argument p after encoding.
+func UserdatasUpdateUserdataDecoderFunc(ctx context.Context, _ *Userdata) mdlserver.UserdatasUpdateUserdataDecoderFunc {
+	return func(reader *multipart.Reader, p **userdatas.UpdateUserdataPayload) error {
+
+		var response *userdatas.UpdateUserdataPayload
+		var res userdatas.UpdateUserdataPayload
+
+		for {
+			part, err := reader.NextPart()
+			if err != nil {
+				if err == io.EOF {
+					break
+				}
+				return userdatas.MakeInternalServerError(errors.Errorf("could not read next part: %w", err))
+			}
+
+			if part.FormName() != "avatar_image" && part.FormName() != "cover_photo" {
+				// Process for other field that's not a file
+
+				buffer, err := ioutil.ReadAll(part)
+				if err != nil {
+					return userdatas.MakeInternalServerError(errors.Errorf("could not process fields: %w", err))
+				}
+				log.WithContext(ctx).Debugf("Multipart process field: %q", part.FormName())
+
+				switch part.FormName() {
+				case "artist_pastelid":
+					res.ArtistPastelID = string(buffer)
+				case "artist_pastelid_passphrase":
+					res.ArtistPastelIDPassphrase = string(buffer)
+				case "biography":
+					value := string(buffer)
+					res.Biography = &value
+				case "categories":
+					value := string(buffer)
+					res.Categories = &value
+				case "facebook_link":
+					value := string(buffer)
+					res.FacebookLink = &value
+				case "location":
+					value := string(buffer)
+					res.Location = &value
+				case "primary_language":
+					value := string(buffer)
+					res.PrimaryLanguage = &value
+				case "native_currency":
+					value := string(buffer)
+					res.NativeCurrency = &value
+				case "realname":
+					value := string(buffer)
+					res.RealName = &value
+				case "twitter_link":
```

Comment:
default case ?

Reply:
i added the default case, just error log
thank you